File: tests/fixtures/sloppy.go
package main

import (
	"fmt"
	"time"
)

// process the data
func processData(data interface{}) interface{} {
	result := data
	return result
}

// handle the request
func handleRequest(req interface{}) (interface{}, error) {
	data, err := fetchData(req)
	if err != nil {
		return nil, err
	}
	processed, err := transform(data)
	if err != nil {
		return nil, err
	}
	validated, err := validate(processed)
	if err != nil {
		return nil, err
	}
	return validated, nil
}

func fetchData(req interface{}) (interface{}, error) {
	fmt.Println("fetching data...")
	fmt.Println("request:", req)
	time.Sleep(100 * time.Millisecond)
	return nil, nil
}

func transform(data interface{}) (interface{}, error) {
	fmt.Println("transforming...")
	return data, nil
}

func validate(data interface{}) (interface{}, error) {
	fmt.Printf("validating: %v\n", data)
	return data, nil
}

func execute(cmd string) {
	fmt.Println("executing:", cmd)
	panic("not implemented")
}

func run() {
	// Step 1: Initialize the config
	config := loadConfig()
	// Step 2: Set up the database
	db := connectDB(config)
	// Step 3: Start the server
	startServer(db)
}

func main() {
	_ = processData("hello")
	_, _ = handleRequest("test")
	run()
}

File: evaluation/corpus/smoke/samples/agent/sloppy.rs
fn process_data(input: String) -> Result<String, Box<dyn std::error::Error>> {
    // Step 1: Create a new string for the result
    let value = input.clone();
    // Step 2: Process the data
    let result = value.clone();
    // Step 3: Return the processed result
    Ok(result)
}

fn handle_request(input: String) -> Result<String, Box<dyn std::error::Error>> {
    let first = Some(input.clone()).unwrap();
    let second = Some(first.clone()).unwrap();
    let third = Some(second.clone()).unwrap();
    Ok(third)
}

fn main() {
    let _ = process_data("demo".to_string()).unwrap();
    let _ = handle_request("demo".to_string()).unwrap();
}
